src/ast/field_analyser.cpp

```diff
@@ -206,22 +229,99 @@ void FieldAnalyser::visit(Predicate &pred)
   pred.expr->accept(*this);
 }
 
-void FieldAnalyser::visit(AttachPoint &ap __attribute__((unused)))
+bool FieldAnalyser::compare_args(std::map<std::string, SizedType> args1,
+                                 std::map<std::string, SizedType> args2)
+{
+  auto pred = [](auto a, auto b) { return a.first == b.first; };
+
+  return args1.size() == args2.size() &&
+         std::equal(args1.begin(), args1.end(), args2.begin(), pred);
+}
+
+bool FieldAnalyser::resolve_args(AttachPoint &ap)
 {
-  if (ap.provider == "kfunc" || ap.provider == "kretfunc")
+  bool kretfunc = ap.provider == "kretfunc";
+  bool lsm = ap.provider == "lsm";
+  std::string func = ap.func;
+
+  if (lsm)
+    func = "bpf_lsm_" + func;
+
+  // load AP arguments into ap_args_
+  ap_args_.clear();
+
+  if (ap.need_expansion)
   {
-    // starting new attach point, clear and load new
-    // variables/arguments for kfunc if detected
+    std::set<std::string> matches;
 
-    ap_args_.clear();
+    try
+    {
+      matches = bpftrace_.find_wildcard_matches(ap);
+    }
+    catch (const WildcardException &e)
+    {
+      std::cerr << e.what() << std::endl;
+      return false;
+    }
+
+    bool first = true;
 
-    if (!bpftrace_.btf_.resolve_args(ap.func,
-                                     ap_args_,
-                                     ap.provider == "kretfunc"))
+    for (auto func : matches)
     {
-      // store/save args for each kfunc ap for later processing
-      bpftrace_.btf_ap_args_.insert({ ap.provider + ap.func, ap_args_ });
+      std::map<std::string, SizedType> args;
 
+      if (!bpftrace_.btf_.resolve_args(func,
+                                       first ? ap_args_ : args,
+                                       kretfunc || lsm))
+      {
+        if (!first && !compare_args(args, ap_args_))
+        {
+          has_mixed_args_ = true;
+          break;
+        }
+      }
+
+      first = false;
+    }
+  }
+  else if (bpftrace_.btf_.resolve_args(func, ap_args_, kretfunc || lsm))
+  {
+    error("Failed to resolve probe arguments", ap.loc);
+    return false;
+  }
+
+  // check if we already stored arguments for this probe
+  auto it = bpftrace_.btf_ap_args_.find(probe_->name());
+
+  if (it != bpftrace_.btf_ap_args_.end())
+  {
+    // we did, and it's different.. save the state and
+    // triger the error if there's args->xxx detected
+    if (!compare_args(it->second, ap_args_))
```

Comment:
It seems this check is too restrictive. For example, `vfs_open(const struct path *path, struct file *file)` and `vfs_write(struct file *file, ...)` have the same arg `struct file *file`, but cannot create the program to access the arg.

```
% sudo ./src/bpftrace -e 'f:vfs_open, f:vfs_write { @ = args->file->f_count; }'
stdin:1:12-24: ERROR: Failed: probe has attach points with mixed arguments
f:vfs_open, f:vfs_write { @ = args->file->f_count; }
           ~~~~~~~~~~~~
```
This works on the current master.